Source/WebCore/rendering/TextBoxPainter.cpp

```diff
@@ -369,6 +369,23 @@ void TextBoxPainter<TextBoxPath>::paintForegroundAndDecorations()
         // Coalesce styles of adjacent marked texts to minimize the number of drawing commands.
         auto coalescedStyledMarkedTexts = StyledMarkedText::coalesceAdjacentWithEqualForeground(styledMarkedTexts);
 
+        if (coalescedStyledMarkedTexts.isEmpty())
+            return;
+
+        bool isHighlights = false;
+
+        for (auto& element : coalescedStyledMarkedTexts) {
+            if (element.type == MarkedText::Type::Highlight) {
+                isHighlights = true;
+                break;
+            }
+        }
+        // Last element is original text with default color, should paint first underneath all marked texts
+        if (auto originalText = coalescedStyledMarkedTexts.last(); isHighlights && originalText.highlightName.isNull()) {
```

Comment:
I'm not sure what other case would apply? From what I tested, it is the last element and spans the range of the entire line of text. I haven't encountered a case where it is not the last element but I'll test it again.

Reply:
In `StyledMarkedText::subdivideAndResolve` I make it so any markedText that does not have a highlightName will be at the end of the vector. Since every single line of text on a webpage is a markedText that does not have a highlightName, they will always be at the end of the vector.